test/unitary/xtypes/collection_types.cpp

```diff
@@ -96,6 +96,37 @@ TEST (CollectionTypes, multi_array)
 
 }
 
+TEST (CollectionTypes, multi_array_constructor)
+{
+
+    ArrayType array_array_array(primitive_type<uint32_t>(), {3, 4, 5});
+
+    EXPECT_EQ(array_array_array.dimension(), 3);
+
```

Comment:
I think it'd be nice to check also here the dimension of the inner types:
`
EXPECT_EQ(array_array_array.content_type().dimension(), 4);
`

Reply:
A rebase was needed, so you must verify the new checks manually :).